### ISSUE TITLE
the android camera2 CameraManager.getCameraCharacteristics() is not the same as ARCore camera.getImageIntrinsics() 

### ISSUE THREAD

**Issue by guagua11** [google-ar/arcore-android-sdk#1633]
### VERSIONS USED
- Android Studio: Android Studio Giraffe | 2022.3.1 Patch 1
- ARCore SDK for Android: com.google.ar:core:1.39.0
- Device manufacturer, model, and O/S: pixel4
- Google Play Services for AR (ARCore):  versionName=1.40.232630283
  On Windows, use: `adb shell pm dump com.google.ar.core | findstr /i "packages: versionName"`
  On macOS, use: `adb shell pm dump com.google.ar.core | egrep -i versionName\|packages:`
- Output of `adb shell getprop ro.build.fingerprint`: Android/aosp_flame/flame:10/QQ3A.200805.001/eng.root.20230829.202716:userdebug/test-keys

### STEPS TO REPRODUCE THE ISSUE
 1.dowload the arcore sdk
 2.load the hello_ar_java demo
 3.on HelloArActivity   function onDrawFrame(SampleRender render)
type 
  Camera camera = frame.getCamera();
 CameraIntrinsics cameraIntrinsics = camera.getImageIntrinsics();
which is the ARCore,s cameraIntrinsics and type 

 CameraManager manager = (CameraManager) this.getSystemService(Context.CAMERA_SERVICE);
        CameraCharacteristics characteristics
                = null;
        try {
            characteristics = manager.getCameraCharacteristics(String.valueOf(android.hardware.Camera.CameraInfo.CAMERA_FACING_BACK));
        } catch (CameraAccessException e) {
            throw new RuntimeException(e);
        }
        float[] intrinsic;
        intrinsic = characteristics.get(CameraCharacteristics.LENS_INTRINSIC_CALIBRATION);
which is the android camera2,s camera intrinsic,but this two intrinsic matrix is not the same as the float[fx,fy,cx,cy,s].
so, can someone tellme what,s wrong with this ,thanks a lot!

**Comment by Totorro35** [comment]
Camera use by ARCore is not always the `android.hardware.Camera.CameraInfo.CAMERA_FACING_BACK`, it's depend of device. You can get the cameraId use by ARCore from the session `session.cameraConfig.cameraId` and the corresponding characteristics with `characteristics = manager.getCameraCharacteristics(session.cameraConfig.cameraId`

**Comment by softlion** [comment]
@Totorro35 
Do you know for what image size the intrinsics returned by getCameraCharacteristics are ?

**Comment by Totorro35** [comment]
getCameraCharacteristics returns the Characteristic of the specify CameraId. `android.hardware.Camera.CameraInfo.CAMERA_FACING_BACK` relates to one physical camera or logical camera (depends on devices). You can find more info about cameraId in MultiCamera API https://developer.android.com/media/camera/camera2/multi-camera.

**Comment by Totorro35** [comment]
The intrinsics returns by getCameraCharacteristics can corresond to different level of zoom, aspect ratio directly managed by the manafucturer of your devices (Samsung provide a cameraId with square shape for example)

**Comment by softlion** [comment]
@Totorro35 yes I think I did try every possible way to use those camera with camera2 and cameraX.

Basically I'm trying to get the same intrinsics as the one returned by ARCore, without using ARCore, in order to use my own stereo build pipeline from multiples images from a selected camera.

Until now I do get the intrinsics from LensIntrinsicCalibration and the sensor pixel size from SensorInfoPreCorrectionActiveArraySize

Then I choose an image capture size, compute scale factors in width and height between this size and the sensor pixel size, and apply those factors to the intrinsics (only on focal length and center)

I'm getting correct horizontal distances, but incorrect vertical distances. So not sure how ARKit does that.

**Comment by softlion** [comment]
@guagua11 did you find a workaround ? It looks like ARCore is computing those intrinsics internally. It may get them from a specific camera, but the returned values are - as you said - not the same as the ones returned from the camera api.

Can't Google open source ARCore SDK now that it is no more a strategic technology ?

**Comment by 15kingben** [comment]
Can you comment more on what is the original problem you are trying to solve? Any content placed in the AR scene should use the ARCore camera's intrinsics

**Comment by softlion** [comment]
I'm wondering how ARcore computes the intrinsics matrix, as it is not the same as what Camera2 api gives for the same camera
